flink-connector-oracle-cdc/src/main/java/com/ververica/cdc/connectors/oracle/table/OracleDeserializationConverterFactory.java

```diff
@@ -0,0 +1,241 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ververica.cdc.connectors.oracle.table;
+
+import org.apache.flink.table.types.logical.LogicalType;
+
+import com.ververica.cdc.debezium.table.DeserializationRuntimeConverter;
+import com.ververica.cdc.debezium.table.DeserializationRuntimeConverterFactory;
+import io.debezium.data.SpecialValueDecimal;
+import io.debezium.data.VariableScaleDecimal;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+
+import java.math.BigDecimal;
+import java.time.ZoneId;
+import java.util.Optional;
+
+/** Used to create {@link DeserializationRuntimeConverterFactory} specified to Oracle. */
+public class OracleDeserializationConverterFactory {
+
+    public static DeserializationRuntimeConverterFactory instance() {
+        return new DeserializationRuntimeConverterFactory() {
+
+            private static final long serialVersionUID = 1L;
+
+            @Override
+            public Optional<DeserializationRuntimeConverter> createUserDefinedConverter(
+                    LogicalType logicalType, ZoneId serverTimeZone) {
+                return wrapNumericConverter(createNumericConverter(logicalType));
+            }
+        };
+    }
+
+    /** Creates a runtime converter which assuming input object is not null. */
+    private static Optional<DeserializationRuntimeConverter> createNumericConverter(
+            LogicalType type) {
+        switch (type.getTypeRoot()) {
+            case BOOLEAN:
+                return createBooleanConverter();
+            case TINYINT:
+                return createByteConverter();
+            case SMALLINT:
+                return createShortConverter();
+            case INTEGER:
+                return createIntegerConverter();
+            case BIGINT:
+                return createLongConverter();
+            case FLOAT:
+                return createFloatConverter();
+            case DOUBLE:
+                return createDoubleConverter();
```

Comment:
I noticed that we've  dealed  `VariableScaleDecimal` in `RowDataDebeziumDeserializeSchema` for `DECIMAL` type.  Should we make these change in  `RowDataDebeziumDeserializeSchema`?  unless debezium using `VariableScaleDecimal` to represent `BOOLEAN` and `TINYINT` only happens `Oracle` Connector, could you help make sure this?

Reply:
I read the `VariableScaleDecimal` source code and found that it is still used in the `NUMERIC` type of `Postgres` (this type is currently mapped to Flink decimal and can be processed normally); 
I think `Oracle` is special because the `FLOAT` type also uses `VariableScaleDecimal` processing;